src/Modules/SharedModule/components/NavBar/NavBar.tsx

```diff
@@ -1,35 +1,39 @@
+import useLocalStorage from "use-local-storage";
 import { useUser } from "../../../../Context/AuthContext";
 import Images from "../../../ImageModule/components/Images/Images";
+import { Link } from "react-router-dom";
 
 export default function NavBar() {
+  const [dark, setDark] = useLocalStorage("dark", false);
```

Comment:
If the package is used for this only, we can use the initialization function syntax in the useState to get the default value in the first render.
```
const [isDarkMode,setIsDarkMode] = React.useState(()=>{
const initialValue = localStorage.getItem("dark");
if(initialValue) return initialValue;
return false;
})
```
This will get the default value from the local storage and if there was no one will return false.
We might also need to parse the value from the local storage as it will be a string not a boolean.